terraform/openstack/cluster.tf

```diff
@@ -1,3 +1,22 @@
+# This terraform creates some resources that are subject to compromises because
+# of some limitations of the magnum driver.
+# 1. Concurrent nodegroups creation requests fired by terraform fail because of
+#    a race condition in the magnum driver. This means that `terraform apply`
+#    must be ran with `--parallelism=1` to avoid this issue.
+#    # https://bugs.launchpad.net/magnum/+bug/2097946
+# 2. The role and labels of the nodegroups don't get propagated to the actual
+#    nodes. This means that we cannot add the node-purpose labels that z2jh
+#    expects and instead we need to rely on the only label that gets added by
+#    the magnum driver and can identify the node, capi.stackhpc.com/node-group.
+#    This in turn, forces us to only offer one machine type per notebook/dask/gpu
+#    for the users.
+#    https://github.com/azimuth-cloud/capi-helm-charts/issues/84
+# 3. The node count and min node count cannot be set to 0 and each nodegroup has
+#    to have at least 1 node.
+#    https://bugs.launchpad.net/magnum/+bug/2098002
+# 4. A default-worker is created apart from the default-controlplane nodegroup
```

Comment:
@jmunroe is dask a requirement for project pythia? if it is, we can leave this be. If not we can disable this by default.

Reply:
Yes, Dask is used within Project Pythia